Building on Windows

Hello,
It seems that PyEDA doesn't easily build on Windows with MSVC, because it uses a C99 feature not available in MSVC.
The error is the following:
```
extension\boolexpr\boolexpr.c(440): error C2057: expected constant expression
extension\boolexpr\boolexpr.c(440): error C2466: cannot allocate an array of constant size 0
extension\boolexpr\boolexpr.c(440): error C2133: 'xs': unknown size
extension\boolexpr\boolexpr.c(449): error C2057: expected constant expression
extension\boolexpr\boolexpr.c(449): error C2466: cannot allocate an array of constant size 0
extension\boolexpr\boolexpr.c(449): error C2133: 'xs': unknown size
[... continues ...]
```

It appears the problem is with the use of variable-length arrays in the `BN_OrN`, `BN_AndN`, and related functions:
```
BX_OrN(size_t n, ...)
{
    struct BoolExpr *xs[n];
    READ_ARGS(n, xs);
    return BX_Or(n, xs);
}
```
This StackOverflow questions seems to indicate this feature will never come to MSVC: https://stackoverflow.com/q/5246900

If I replace the 8ish uses of VLAs in `boolexpr.c` with calls to `alloca()`, it appears to allow PyEDA to build easily on Windows. I have the changes in a branch and can open a pull request if you like? What do you think about this change?

Here's the diff: https://github.com/alexwebr/pyeda/commit/a07acb0f09ac020b906976df61179ea049a6331a

Happy to discuss, and thanks for the great software!

--- a/extension/boolexpr/boolexpr.c
+++ b/extension/boolexpr/boolexpr.c
@@ -20,6 +20,11 @@
 #include "share.h"
 #include "util.h"
 
+#ifdef _WIN32
+/* https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/alloca */
+#define alloca _alloca
+#endif
+
 
 #define READ_ARGS(n, xs) \
 do { \
@@ -437,7 +442,7 @@ BX_ITE(struct BoolExpr *s, struct BoolExpr *d1, struct BoolExpr *d0)
 struct BoolExpr *
 BX_OrN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_Or(n, xs);
 }
@@ -446,7 +451,7 @@ BX_OrN(size_t n, ...)
 struct BoolExpr *
 BX_NorN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_Nor(n, xs);
 }
@@ -455,7 +460,7 @@ BX_NorN(size_t n, ...)
 struct BoolExpr *
 BX_AndN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_And(n, xs);
 }
@@ -464,7 +469,7 @@ BX_AndN(size_t n, ...)
 struct BoolExpr *
 BX_NandN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_Nand(n, xs);
 }
@@ -473,7 +478,7 @@ BX_NandN(size_t n, ...)
 struct BoolExpr *
 BX_XorN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_Xor(n, xs);
 }
@@ -482,7 +487,7 @@ BX_XorN(size_t n, ...)
 struct BoolExpr *
 BX_XnorN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_Xnor(n, xs);
 }
@@ -491,7 +496,7 @@ BX_XnorN(size_t n, ...)
 struct BoolExpr *
 BX_EqualN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_Equal(n, xs);
 }
@@ -500,7 +505,7 @@ BX_EqualN(size_t n, ...)
 struct BoolExpr *
 BX_UnequalN(size_t n, ...)
 {
-    struct BoolExpr *xs[n];
+    struct BoolExpr **xs = alloca(sizeof(struct BoolExpr) * n);
     READ_ARGS(n, xs);
     return BX_Unequal(n, xs);
 }

--- a/extension/boolexpr/boolexpr.c
+++ b/extension/boolexpr/boolexpr.c
@@ -23,6 +23,8 @@
 #ifdef _WIN32
 /* https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/alloca */
 #define alloca _alloca
+#else
+#include <alloca.h>
 #endif
 
 

